### ISSUE TITLE
Add an analyzer for equality checks with hash sets + a comparer func

### ISSUE THREAD

**Issue by bradwilson** [xunit/xunit#2840]
Now that https://github.com/xunit/xunit/issues/2838 has been resolved, we should write an analyzer which flags as an error any time the user writes an equality comparison using either of these two functions:

```csharp
void Equal<T>(IEnumerable<T> expected, IEnumerable<T> actual, Func<T, T, bool> comparer)
void NotEqual<T>(IEnumerable<T> expected, IEnumerable<T> actual, Func<T, T, bool> comparer)
```

when `expected` and `actual` are know to be sets (that is, something which implements either `ISet<>` or `IReadOnlySet<>`).

**Comment by etherfield** [comment]
I'm gonna help with it.

**Comment by bradwilson** [comment]
Available in `1.9.0-pre.4` https://xunit.net/docs/using-ci-builds
